### PR TITLE
Replaced hard-coded dimensions by dimens resource values (Fixed #1120)

### LINKED ISSUE #1120: Replace hard-coded dimensions in layout resource files by dimens resource values
**Actual Behaviour**

Currently, there some dimensions, say dimensions of skill detail image as in the screenshot provided below, which have been hard-coded. 

**Expected Behaviour**

Dimensions should be referenced from the dimens.xml resource file rather than hard-coding them.

**Example screenshot of the issue**

dimen

**Would you like to work on the issue?**

Yes.

### PR DESCRIPTION
**Pull request by arundhati24**
Fixes #1120

Changes:  There were a number of dimensions that were hard-coded in the layout
resource files. Such dimensions shall now be referenced from the dimen
resource values specified in dimens.xml.

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -49,6 +48,7 @@ android {
         abortOnError false
         disable 'InvalidPackage'
     }
+    buildToolsVersion '27.0.3'
```
Comment (iamareebjamal) [inline app/build.gradle]: This should not be changed here

File: app/config.gradle [hunk 2]
```diff
@@ -7,9 +7,9 @@ ext {
     //Libraries
     playServicesVersion = '10.2.0'
     supportLibraryVersion = '25.4.0'
-    leakCanaryVersion = '1.5.1'
+    leakCanaryVersion = '1.5.4'
     mpAndroidChartVersion = 'v2.2.5'
-    butterKnifeVersion = '8.4.0'
+    butterKnifeVersion = '8.8.1'
```
Comment (iamareebjamal) [inline app/config.gradle]: Unrelated to the issue

File: app/src/androidTest/java/org/fossasia/susi/ai/login/LoginActivityTest.kt [hunk 3]
```diff
@@ -59,23 +59,28 @@ class LoginActivityTest {
...
 
         // checks if checkbox is present
+        onView(withId(R.id.customer_server)).perform(scrollTo())
         onView(withId(R.id.customer_server)).check(matches(isDisplayed()))
 
         // checks if sign up button is present
-        onView(withId(R.id.sign_up)).perform(scrollTo())
-        onView(withId(R.id.sign_up)).check(matches(isDisplayed()))
+        onView(withId(R.id.login_activity_sign_up)).perform(scrollTo())
+        onView(withId(R.id.login_activity_sign_up)).check(matches(isDisplayed()))
```
Comment (iamareebjamal) [inline app/src/androidTest/java/org/fossasia/susi/ai/login/LoginActivityTest.kt]: Unrelated

File: app/src/androidTest/java/org/fossasia/susi/ai/skills/SkillsActivityTest.kt [hunk 4]
```diff
@@ -54,7 +54,7 @@ class SkillsActivityTest {
     fun testUIViews() {
         Log.d(TAG, "running testUIViews..")
 
-        // checks if frame layout is displayed or not
+        //checks if frame layout is displayed or not
```
Comment (iamareebjamal) [inline app/src/androidTest/java/org/fossasia/susi/ai/skills/SkillsActivityTest.kt]: Unrelated

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt [hunk 5]
```diff
@@ -117,7 +117,7 @@ class LoginActivity : AppCompatActivity(), ILoginView {
     }
 
     fun signUp() {
-        sign_up.setOnClickListener { startActivity(Intent(this@LoginActivity, SignUpActivity::class.java)) }
+        login_activity_sign_up.setOnClickListener { startActivity(Intent(this@LoginActivity, SignUpActivity::class.java)) }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: Unrelated

File: app/src/main/res/layout-land/activity_forgot_password.xml [hunk 6]
```diff
@@ -60,7 +60,7 @@
             android:id="@+id/reset_button"
             android:layout_width="@dimen/activity_landscape_width"
             android:layout_height="wrap_content"
-            android:layout_marginTop="16dp"
+            android:layout_marginTop="@dimen/margin_button_large"
```
Comment (iamareebjamal) [inline app/src/main/res/layout-land/activity_forgot_password.xml]: Rather than defining `margin_button_large`, only use set of spacings like `margin_large`

File: build.gradle [hunk 7]
```diff
@@ -7,11 +7,11 @@ buildscript {
...
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.0.1'
+        classpath 'com.android.tools.build:gradle:3.1.2'
         classpath 'com.jfrog.bintray.gradle:gradle-bintray-plugin:1.2'
         classpath 'com.github.dcendents:android-maven-gradle-plugin:1.4.1'
-        classpath "io.realm:realm-gradle-plugin:3.5.0"
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.1.3"
+        classpath "io.realm:realm-gradle-plugin:3.7.2"
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.2.30"
```
Comment (iamareebjamal) [inline build.gradle]: Unneeded

File: gradle/wrapper/gradle-wrapper.properties [hunk 8]
```diff
@@ -1,6 +1,6 @@
-#Tue Dec 12 15:49:37 IST 2017
+#Mon Apr 30 11:20:13 IST 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-4.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.4-all.zip
```
Comment (iamareebjamal) [inline gradle/wrapper/gradle-wrapper.properties]: Unneeded

File: app/src/androidTest/java/org/fossasia/susi/ai/login/LoginActivityTest.kt (hunk 3 above)
Comment (arundhati24) [reply app/src/androidTest/java/org/fossasia/susi/ai/login/LoginActivityTest.kt]: @iamareebjamal As the UI tests were failing that time and on going through the logs what I understood that time was that there was a trouble differentiating between the id "sign_up" which is present in both the Login and SignUp activity. Hence, I tried changing it for the Login Activity tests and it did pass the UI tests in Android studio. But, now I see that this doesn't turn up, there are some other issues. I will change this back to the original, as I have done in one of the other PRs.

File: app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt (hunk 5 above)
Comment (arundhati24) [reply app/src/main/java/org/fossasia/susi/ai/login/LoginActivity.kt]: @iamareebjamal As the UI tests were failing that time and on going through the logs what I understood that time was that there was a trouble differentiating between the id "sign_up" which is present in both the Login and SignUp activity. Hence, I tried changing it for the Login Activity tests and it did pass the UI tests in Android studio. But, now I see that this doesn't turn up, there are some other issues. I will change this back to the original, as I have done in one of the other PRs.

File: app/build.gradle (hunk 1 above)
Comment (arundhati24) [reply app/build.gradle]: @iamareebjamal As each time this requires to be updated, should I open a new issue to change it?

File: app/src/main/res/layout-land/activity_forgot_password.xml (hunk 6 above)
Comment (arundhati24) [reply app/src/main/res/layout-land/activity_forgot_password.xml]: Sure @iamareebjamal but here, margin_large was already defined with a value of 10dp hence I had to use a name like that.

File: app/build.gradle (hunk 1 above)
Comment (iamareebjamal) [reply app/build.gradle]: Yes, open a new issue to upgrade all dependencies

File: gradle/wrapper/gradle-wrapper.properties (hunk 8 above)
Comment (arundhati24) [reply gradle/wrapper/gradle-wrapper.properties]: @iamareebjamal Builds fail without this update.. It won't be able to pass CircleCI tests. Please suggest, what should be done?

image

Comment (iamareebjamal) [reply gradle/wrapper/gradle-wrapper.properties]: Then a PR updating dependencies should be merged before this. This cannot be added in this PR

Comment (arundhati24) [reply gradle/wrapper/gradle-wrapper.properties]: Sure.  I will do this asap.

Comment (arundhati24) [reply gradle/wrapper/gradle-wrapper.properties]: @iamareebjamal As all dependencies have been updated now, this PR too has passed all checks. It would be great if you could please review this too.

**Comment by iamareebjamal** [comment]
@arundhati24 There are still many unrelated changes in this PR

**Comment by arundhati24** [comment]
@iamareebjamal I will fix that asap.

**Comment by arundhati24** [comment]
@iamareebjamal I removed those unrelated changes.

**Comment by arundhati24** [comment]
@iamareebjamal Thanks and I completely agree with you. I too faced an issue when I had to name those dimensions. But, should I implement that change in this PR itself or should I open a new issue to use standard naming conventions in accordance with Material Design Guidelines because that requires a lot of change throughout the dimens.xml as well as the layout resource files while this PR only deals with replacement of hard-coded dimens. 
I would proceed as you direct.

**Comment by arundhati24** [comment]
@batbrain7 I will do that for sure. This PR needs to be delayed for some time as of now.

**Comment by mariobehling** [comment]
@arundhati24 This PR has been delayed now for a long time. During GSoC there was time to implement proper naming conventions and get things ready for this PR. What is the status now? Please finalize this PR.

**Comment by arundhati24** [comment]
As #1666 takes care of the subject of this PR, I am closing this now.
